src/js/client/controls.js

```diff
@@ -0,0 +1,150 @@
+        switch (event.keyCode) {
+            case 27: // escape
+                that.toggleMenu();
+                break;
+        }
+
+        // TODO:  Eventually, we may want to take over control of "natural" arrow key handling using event.preventDefault()
+    };
+
+    gpii.express.user.frontend.controls.handleToggleKeys = function (that, event) {
+        switch (event.keyCode) {
+            case 27: // escape
+                that.toggleMenu();
+                break;
+            case 13: // enter
+                that.toggleMenu();
+                break;
+        }
+    };
+
+    gpii.express.user.frontend.controls.handleLogoutKeys = function (that, event) {
+        switch (event.keyCode) {
+            case 13: // enter
+                that.submitForm(event);
+                break;
+        }
+    };
+
+    gpii.express.user.frontend.controls.toggleMenu = function (that) {
+        var toggle = that.locate("toggle");
+        var menu   = that.locate("menu");
+
+        if ($(menu).is(":hidden")) {
+            menu.show();
+            menu.focus();
+        }
+        else {
+            menu.hide();
+            toggle.focus();
+        }
+    };
+
+    fluid.defaults("gpii.express.user.frontend.controls", {
+        gradeNames: ["gpii.templates.templateFormControl"],
+        container:  ".controls-viewport",
+        ajaxOptions: {
+            type:     "GET",
+            url:      "/api/user/logout"
+        },
+        templates: {
+            initial: "controls-viewport",
+            success: "common-success",
+            error:   "common-error"
+        },
+        rules: {
+            successResponseToModel: {
+                "":        "notfound",
+                user: {
+                    literalValue: false
+                }
+            }
+        },
+        selectors: {
+            initial:  "",
+            success:  ".controls-message",
+            error:    ".controls-message",
+            controls: ".user-controls",
+            menu:     ".user-menu",
+            logout:   ".user-menu-logout",
+            toggle:   ".user-controls-toggle"
+        },
+        model: {
+            user: null
+        },
+        modelListeners: {
+            user: {
+                func:          "{that}.renderInitialMarkup",
+                excludeSource: "init"
+            }
+        },
+        invokers: {
+            toggleMenu: {
+                funcName: "gpii.express.user.frontend.controls.toggleMenu",
+                args:     [ "{that}"]
+            },
+            handleMenuKeys: {
+                funcName: "gpii.express.user.frontend.controls.handleMenuKeys",
+                args:     [ "{that}", "{arguments}.0"]
+            },
+            handleToggleKeys: {
+                funcName: "gpii.express.user.frontend.controls.handleToggleKeys",
+                args:     [ "{that}", "{arguments}.0"]
+            }
+        },
+        listeners: {
+            onMarkupRendered: [
```

Comment:
Supply these with namespaces rather than sticking in a large array so that they can easily be overridden individually - 

e.g. 
"onMarkupRendered.logout": {
                    "this":   "{that}.dom.logout",
                    "method": "click",
                    "args":   "{that}.submitForm"
                }

incidentally - why does a click on the logout control submit the form?